### PR TITLE
8283404: [macos] a11y : Screen magnifier does not show JMenu name

### PR DESCRIPTION
**Pull request by kumarabhi006**
It seems that the accessibility role returned by the MenuAccessibility class is not correct for top level menu.

Added a condition check to return the accessibility role as `NSAccessibilityMenuBarItemRole` in stead of `NSAccessibilityMenuRole` for top level menu whose parent is `menubar`. 

The solution works correctly for screen magnifier as well as voice over. 

A manual test case is added to verify the fix.

Note: Tried implementing few a11y methods to fix but for top level menu with `NSAccessibilityMenuRole` doesn't return the a11y label.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8283404](https://bugs.openjdk.org/browse/JDK-8283404): [macos] a11y : Screen magnifier does not show JMenu name

### Reviewers
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**)
 * @M4ximumPizza (no known openjdk.org user name / role)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13271/head:pull/13271` \
`$ git checkout pull/13271`

Update a local copy of the PR: \
`$ git checkout pull/13271` \
`$ git pull https://git.openjdk.org/jdk.git pull/13271/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13271`

View PR using the GUI difftool: \
`$ git pr show -t 13271`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13271.diff">https://git.openjdk.org/jdk/pull/13271.diff</a>

</details>

### CODE REVIEW

File: test/jdk/javax/accessibility/JMenu/TestJMenuScreenMagnifier.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import javax.swing.JMenuItem;
+import javax.swing.SwingUtilities;
+
+public class TestJMenuScreenMagnifier {
+    private static JFrame frame;
+    private static final String INSTRUCTIONS =
+            "1) Enable Screen magnifier on theMac \n\n" +
+                "System Preference -> Accessibility -> Zoom -> " +
+                "Select ( Enable Hover Text) \n\n" +
+            "2) Run with -Djavax.accessibility.screen_magnifier_present=true" +
```
Comment (mrserb) [inline test/jdk/javax/accessibility/JMenu/TestJMenuScreenMagnifier.java:48]: What the user should "Run" at this step?

Comment (kumarabhi006) [reply test/jdk/javax/accessibility/JMenu/TestJMenuScreenMagnifier.java:48]: This step is not required. Copy+paste mistake from JBS description.
I will update the test instruction.

Comment (kumarabhi006) [reply test/jdk/javax/accessibility/JMenu/TestJMenuScreenMagnifier.java]: > What the user should "Run" at this step?

Updated the test instruction.

**Review by M4ximumPizza — APPROVED** [review]

**Pull request closed without merging**